common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -2177,6 +2177,12 @@
     ],
     "sqlState" : "42001"
   },
+  "INVALID_EXTERNAL_TYPE" : {
+    "message" : [
+      "<actualType> is not a valid external type for schema of <expectedType> at <childExpression>."
```

Comment:
```suggestion
      "The external type <externalType> is not valid for the type <type> at the expression <expr>."
```

Reply:
But I would like to mention that `QueryExecutionErrors::invalidExternalTypeError` will look like this:

```
  def invalidExternalTypeError(
      actualType: String,
      expectedType: DataType,
      childExpression: Expression): SparkRuntimeException = {
    new SparkRuntimeException(
      errorClass = "INVALID_EXTERNAL_TYPE",
      messageParameters = Map(
        "externalType" -> actualType,
        "type" -> quoteByDefault(expectedType.simpleString),
        "expr" -> toSQLExpr(childExpression)
      )
    )
  }
```

We have `actualType` => `externalType`, `expectedType` => `type`, `childExpression` => `exp` renaming because at least `type` is a keyword in Scala.

Reply:
sgtm, could you use `toSQLType`:
```scala
    new SparkRuntimeException(
      errorClass = "INVALID_EXTERNAL_TYPE",
      messageParameters = Map(
        "externalType" -> actualType,
        "type" -> toSQLType(expectedType),
        "expr" -> toSQLExpr(childExpression)
      )
    )
```